### DISCUSSION TITLE
Cypress doesn't navigate to some address

### DISCUSSION THREAD

**Discussion by T-Auto-1** [cypress-io/cypress#25850] [Questions and Help]
When I run `cy.visit()` to navigate to a specific address this message appears in the browser (not the cypress command log):  **[the address] did not send any data.**
If I open a new tab on the browser that cypress has started and I navigate to the address manually I also see this message: **ERR_EMPTY_RESPONSE**

For other addresses, `cy.visit()` works fine.
When I navigate to the same address by using a separate browser (not started by cypress), it works fine. Also when I navigate to this address with Playwright framework, it works too.

Setting `chromeWebSecurity` to false doesn't help, neither using other built-in browsers.
See screenshot. The message is in the center of the page in the black rectangle. The reason I have put these rectangles is because of company policy.
![cy](https://user-images.githubusercontent.com/121883042/219379055-b4a7fc3f-e839-4a41-a2f4-c923b73442eb.png)

How can I solve this?

**Comment by portyu9** [comment]
Hi @T-Auto-1 

Since the URL also fails when entered manually in a new tab of the **browser launched by Cypress**, this is not a `cy.visit()` issue. It is a network/proxy configuration difference in the Cypress-launched browser.

Check what proxy Cypress is using:

```bash
npx cypress info
```

If the target is an internal/company URL that should bypass the corporate proxy, add it to `NO_PROXY` before starting Cypress:

```bash
# macOS/Linux
export NO_PROXY="your-company-domain.com"
npx cypress open
```

```powershell
# Windows PowerShell
$env:NO_PROXY="your-company-domain.com"
npx cypress open
```

If your company uses a PAC file, Cypress does not support PAC directly. Configure the actual HTTP/HTTPS proxy explicitly instead:

```bash
HTTP_PROXY=http://proxy.company.com \
HTTPS_PROXY=http://proxy.company.com \
npx cypress open
```

`chromeWebSecurity: false` and increasing `pageLoadTimeout` will not fix `ERR_EMPTY_RESPONSE`.

The fix is to make Cypress use the same **proxy/bypass route** that your normal browser uses.

**Reply by portyu9** [reply]
@T-Auto-1 Please mark this answer as accepted if it helped, thank you.
